include/zephyr/spinlock.h

```diff
@@ -29,6 +29,35 @@ extern "C" {
  * @{
  */
 
+/**
+ * @brief Execute a code block with the given spinlock held, automatically
+ * releasing it in the end.
+ *
+ * @details Example usage:
+ *
+ * @code{.c}
+ * WITH_SPINLOCK(&mylock) {
+ *   ...execute protected statements with the lock held...
+ * }
+ * @endcode
+ *
+ * This basically expands to:
+ *
+ * @code{.c}
+ * for (k_spinlock_key_t key = k_spin_lock(&mylock); k_spin_unlock(&mylock, key); ...) {
```

Comment:
Minor detail: The expansion is incorrect. The compare element in a for-loop is executed before we actually enter the body of the for-loop. As a result, there is a discrepancy with the claim above on line 40.

Reply:
@peter-mitsis Should be correct now. Could you have another look?